### PR TITLE
Fix integration tests for OSX

### PR DESCRIPTION
**Pull request by 2opremio**
My impressive first pull request. I tried to make architecture independent changes, but can somebody test it in Linux before merging?

### CODE REVIEW

**Comment by 2opremio** [comment]
Oh, I see you are using Travis, I guess that implicitly checks I didn't break anything in Linux.

**Comment by bboreham** [comment]
If you do the change in a branch in weaveworks/weave then CircleCI will run even more tests.

**Comment by 2opremio** [comment]
@bboreham Thanks, should I recreate the pull request in a weaveworks/weave branch?

**Comment by bboreham** [comment]
It's not essential.  Next time.

**Comment by 2opremio** [comment]
@bboreham Is there some sort of  "review process"? Do I need a +1/LGTM from a team member to go ahead and merge?

**Comment by bboreham** [comment]
Per the [development process](https://github.com/weaveworks/weave/wiki/WorkingOnWeave) (which I just updated), the reviewer merges.

I'll act as reviewer.

File: test/500_weave_multi_cidr_test.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ assert_zone_records() {
 
     records=$(weave_on $HOST status | grep "^$CID") || true
     # Assert correct number of records exist
-    assert "echo $records | grep -oE '\b([0-9]{1,3}\.){3}[0-9]{1,3}\b' | wc -l" $#
+    assert "echo $records | grep -oE '\b([0-9]{1,3}\.){3}[0-9]{1,3}\b' | wc -l | tr -d '[[:space:]]'" $#
```
Comment (bboreham) [inline test/500_weave_multi_cidr_test.sh]: Are two square-brackets necessary?  My reading suggests a single pair will do.

File: test/config.sh [hunk 2]
```diff
@@ -60,16 +60,16 @@ upload_executable() {
 remote() {
     rem=$1
     shift 1
-    "$@" > >(while read line; do echo -e "\e[0;34m$rem>\e[0m $line"; done)
+    "$@" > >(while read line; do echo -e $"\e[0;34m$rem>\e[0m $line"; done)
```
Comment (bboreham) [inline test/config.sh]: Why do we want this `$` ?  (I read the man-page and still don't know)
"A double-quoted string preceded by a dollar sign ($) will cause the string to be translated according to  the  current  locale."

File: test/500_weave_multi_cidr_test.sh (hunk 1 above)
Comment (2opremio) [inline test/500_weave_multi_cidr_test.sh]: You are right, they are not.

File: test/config.sh (hunk 2 above)
Comment (2opremio) [inline test/config.sh]: You are right again, they should be single quotes, but it also works with double-quotes for some reason.

For refernce: OSX's echo doesn't interpret escape characters (`\e`). By prepending `$`to a string, bash will 'encode' the escaped characters before passing them on to echo. 

```
   Words  of  the  form  $'string' are treated specially.  The word expands to string, with
   backslash-escaped characters replaced as specified by the ANSI  C  standard.   Backslash
   escape sequences, if present, are decoded as follows:
          \a     alert (bell)
          \b     backspace
          \e     an escape character
          \f     form feed
          \n     new line
          \r     carriage return
          \t     horizontal tab
          \v     vertical tab
          \\     backslash
          \'     single quote
          \nnn   the  eight-bit  character whose value is the octal value nnn (one to three
                 digits)
          \xHH   the eight-bit character whose value is the hexadecimal value  HH  (one  or
                 two hex digits)
          \cx    a control-x character

   The expanded result is single-quoted, as if the dollar sign had not been present.

   A  double-quoted string preceded by a dollar sign ($) will cause the string to be trans‐
   lated according to the current locale.  If the current locale is C or POSIX, the  dollar
   sign  is  ignored.  If the string is translated and replaced, the replacement is double-
   quoted.
```

**Comment by 2opremio** [comment]
@bboreham I addressed your comments and rebased with master

**Pull request merged**